src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -2992,6 +2992,183 @@ public static Vector512<float> Invoke(Vector512<float> x)
 #endif
         }
 
+        /// <summary>MathF.Sinh(x)</summary>
+        private readonly struct SinhOperator : IUnaryOperator
+        {
+            public static float Invoke(float x) => MathF.Sinh(x);
+            public static Vector128<float> Invoke(Vector128<float> x) => (ExpOperator.Invoke(x) - ExpOperator.Invoke(-x)) / Vector128.Create(2f);
```

Comment:
Is there anything better to do than the standard formula? I didn't see a sinh implementation in the amd code.

Reply:
Don't know about other implementations but exp(x)-exp(y) is not good numerically when x and y are similar (around 0).

Better way to calculate would perhaps be the form: sinh(x) = (exp(2*x) - 1) / (2 * exp(x)) where we use expm1 for the numerator (if it exists as an operator here).

In other words expm1(2*x) / (2*x);

Reply:
`aocl-libm` doesn't have a vectorized version, but does have a scalar version which could be vectorized: https://github.com/amd/aocl-libm-ose/blob/aocl-4.1/src/optimized/sinhf.c

-- Notably, the scalar version of coshf looks more accurate, where-as the vectorized version looks like it estimates a little bit more for speed. The branches are relegated to special inputs (like NaN) which should be rare, and the handling of very large/small values (which should also be rare). We should be able to, in general, isolate those "special cases" to a single branch so the core stays fast for the main cases and the `ConditionalSelect` is rarely ever needed (much as we've done for other algorithms so far).